lib/pedicel/validator.rb

```diff
@@ -80,59 +116,51 @@ def self.messages
 
       required(:deviceManufacturerIdentifier).filled(:str?, :hex?)
 
-      required(:paymentDataType).filled(:str?, included_in?: ['3DSecure', 'EMV'])
+      required(:paymentDataType).filled(:str?, included_in?: %w[3DSecure EMV])
 
       required(:paymentData).schema do
         optional(:onlinePaymentCryptogram).filled(:str?, :base64?)
-        optional(:eciIndicator).filled(:str?)
+        optional(:eciIndicator).filled(:str?, :eci?)
 
         optional(:emvData).filled(:str?, :base64?)
         optional(:encryptedPINData).filled(:str?, :hex?)
       end
 
-      rule('consistent paymentDataType and paymentData': [:paymentDataType, [:paymentData, :onlinePaymentCryptogram]]) do |t, cryptogram|
+      rule('paymentDataType affects paymentData': [:paymentDataType, [:paymentData, :onlinePaymentCryptogram]]) do |t, cryptogram|
         t.eql?('3DSecure') > cryptogram.filled?
       end
     end
 
-    # Pedicel::Validator::TokenDataSchema.call(
-    #   applicationPrimaryAccountNumber: '1234567890123',
-    #   applicationExpirationDate: '101112',
-    #   currencyCode: '123',
-    #   transactionAmount: 12.34,
-    #   cardholderName: 'asdf',
-    #   deviceManufacturerIdentifier: 'adsf',
-    #   paymentDataType: 'asdf',
-    # )
-
-    def self.validate_token(token, now: Time.now)
+    def self.validate_token(token)
       validation = TokenSchema.call(token)
 
-      raise TokenFormatError, validation.hints.map{|key,msg| "#{key} #{msg}"}.join(', and ') unless validation.errors.empty?
+      raise TokenFormatError, format_errors(validation) if validation.failure?
 
       true
     end
 
-    def self.valid_token?(token, now: Time.now)
-      validate_token(token, now: now) rescue false
+    def self.valid_token?(token)
+      validate_token(token)
+    rescue
```

Comment:
An unconditional `rescue` hurts my eyes, however, it might be justified? It will probably make some errors fly under the radar or make debugging difficult at some point.

Reply:
I remember having seen more of them earlier in this PR.

Reply:
Exactly. My fear is that it hides unexpected problems, e.g. typos...

```
$ irb
irb(main):001:0> begin
irb(main):002:1*   invalid ruby
irb(main):003:1> rescue
irb(main):004:1>   puts 42
irb(main):005:1> end
42
```

Reply:
A spec that `expect().to be true` should catch that kind of problems. But, I see your point. And in this case there's actually only one error that should be caught. Will fix!